src/NUnitFramework/framework/Assert.cs

```diff
@@ -320,8 +320,18 @@ private static void ReportFailure(string message)
             result.RecordTestCompletion();
 
             // If we are outside any multiple assert block or multiple asserts disabled, then throw
-            if (TestExecutionContext.CurrentContext.MultipleAssertLevel == 0 || (TestExecutionContext.CurrentContext.DisableMultipleAssertsUnderDebugger && Debugger.IsAttached))
-                throw new AssertionException(result.Message);
+            var debugMultiple = TestExecutionContext.CurrentContext.ThrowOnEachFailureUnderDebugger && Debugger.IsAttached;
+            if (TestExecutionContext.CurrentContext.MultipleAssertLevel == 0 || debugMultiple)
+            {
+                try
+                {
+                    throw new AssertionException(result.Message);
+                }
+                catch (AssertionException) when (debugMultiple && TestExecutionContext.CurrentContext.MultipleAssertLevel > 0)
```

Comment:
@manfred-brands This looks functionally correct to me, but barring time to run this myself I may need to defer to your recent foray on multiple assertion exceptions (if you have time yourself, of course) 

@MaceWindu out of curiosity, why not have used `Debugger.Break()`?

Reply:
Now I think it would be better to use Debugger.Break here

Pros:
- simplier/less code
- no need for user to setup debugger to break on handled exceptions

Cons:
- `result.Message` not explicitly available, user need to inspect it
- as @manfred-brands said - it will break always even if user don't want it. But in that case he shouldn't have enabled option I think
- need to rename setting again 😄 

> Not all the other properties are copied either.

That's why I didn't do it - I've checked how other settings handled

Reply:
Thanks for the explanation @MaceWindu 😃 
It does make sense to me to prioritize user friendliness on a setting like this, which it sounds like was how you approached it too